From 182ece43e4a17f9ddd56fcf256a4a7ebb0ae8a31 Mon Sep 17 00:00:00 2001
From: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
Date: Mon, 12 Jan 2026 16:50:08 +0200
Subject: [PATCH 113/135] arm64: dts: rockchip: Add frl-enable-gpios to
 rk3576-rock-4d

In preparation to support HDMI 2.1 FRL operating mode, make use of the
GPIO2_B0 line (labeled HDMI_TX_ON_H in the schematics) to control the
voltage bias for HDMI data lines.

The GPIO line is asserted when operating in FRL mode and deasserted for
TMDS.

Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts b/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts
index b0d69908d..e36c50c51 100644
--- a/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3576-rock-4d.dts
@@ -348,6 +348,7 @@ &gpu {
 };
 
 &hdmi {
+	frl-enable-gpios = <&gpio2 RK_PB0 GPIO_ACTIVE_LOW>;
 	status = "okay";
 };
 
-- 
2.47.3

